### PR TITLE
refactor(radio): Annotate mdc-radio for closure

### LINKED ISSUE #340: Annotate mdc-radio for closure

### PR DESCRIPTION
**Pull request by lynnjepsen**
Resolves #340

### CODE REVIEW

**Comment by lynnjepsen** [comment]
This is a copy of #886, but with the changes I originally requested. @nckh hasn't replied to the comments for more than a week, and we generally close PRs that are non-responsive, but I didn't want to lose this valuable work before closing out the PR.

File: packages/mdc-checkbox/adapter.js [hunk 1]
```diff
@@ -52,21 +54,11 @@ export default class MDCCheckboxAdapter {
   /** @param {!EventListener} handler */
   deregisterChangeHandler(handler) {}
 
-  /** @return {InputElementState} */
+  /** @return {SelectionControlState} */
```
Comment (acdvorak) [inline packages/mdc-checkbox/adapter.js]: `{!SelectionControlState}`

File: packages/mdc-checkbox/index.js [hunk 2]
```diff
@@ -34,12 +34,12 @@ export class MDCCheckbox extends MDCComponent {
   }
 
   /**
-   * @return {InputElementState|undefined}
+   * @return {SelectionControlState|undefined}
```
Comment (acdvorak) [inline packages/mdc-checkbox/index.js]: `{SelectionControlState|undefined}` -> `{?SelectionControlState}`

`querySelector()` returns `null`, not `undefined`.

File: packages/mdc-checkbox/index.js [hunk 3]
```diff
@@ -34,12 +34,12 @@ export class MDCCheckbox extends MDCComponent {
...
    * @private
    */
   get nativeCb_() {
     const {NATIVE_CONTROL_SELECTOR} = MDCCheckboxFoundation.strings;
-    const cbEl = /** @type {InputElementState|undefined} */ (
+    const cbEl = /** @type {SelectionControlState|undefined} */ (
```
Comment (acdvorak) [inline packages/mdc-checkbox/index.js]: `{SelectionControlState|undefined}` -> `{?SelectionControlState}`

`querySelector()` returns `null`, not `undefined`.

File: packages/mdc-checkbox/foundation.js [hunk 4]
```diff
@@ -47,7 +48,7 @@ export default class MDCCheckboxFoundation extends MDCFoundation {
       deregisterAnimationEndHandler: (/* handler: EventListener */) => {},
       registerChangeHandler: (/* handler: EventListener */) => {},
       deregisterChangeHandler: (/* handler: EventListener */) => {},
-      getNativeControl: () => /* InputElementState */ {},
+      getNativeControl: () => /* SelectionControlState */ {},
```
Comment (acdvorak) [inline packages/mdc-checkbox/foundation.js]: `!SelectionControlState`

File: packages/mdc-radio/foundation.js [hunk 5]
```diff
@@ -14,38 +14,52 @@
...
+import {SelectionControlState} from '@material/base/selection-control';
+import MDCRadioAdapter from './adapter';
+/* eslint-enable no-unused-vars */
 import {cssClasses, strings} from './constants';
 
+/**
+ * @extends {MDCFoundation<!MDCRadioAdapter>}
+ */
 export default class MDCRadioFoundation extends MDCFoundation {
+  /** @return enum{cssClasses} */
```
Comment (acdvorak) [inline packages/mdc-radio/foundation.js]: `enum{cssClasses}` -> `enum {cssClasses}` (add a space)

File: packages/mdc-radio/foundation.js [hunk 6]
```diff
@@ -14,38 +14,52 @@
...
   static get cssClasses() {
     return cssClasses;
   }
 
+  /** @return enum{strings} */
```
Comment (acdvorak) [inline packages/mdc-radio/foundation.js]: `enum{strings}` -> `enum {strings}` (add a space)

File: packages/mdc-radio/foundation.js [hunk 7]
```diff
@@ -14,38 +14,52 @@
...
 
+  /** @return {!MDCRadioAdapter} */
   static get defaultAdapter() {
-    return {
+    return /** @type {!MDCRadioAdapter} */ ({
       addClass: (/* className: string */) => {},
       removeClass: (/* className: string */) => {},
-      getNativeControl: () => /* HTMLInputElement */ {},
-    };
+      getNativeControl: () => /* SelectionControlState */ {},
```
Comment (acdvorak) [inline packages/mdc-radio/foundation.js]: `!SelectionControlState`

File: packages/mdc-radio/index.js [hunk 8]
```diff
@@ -84,15 +102,24 @@ export class MDCRadio extends MDCComponent {
     return new MDCRipple(this.root_, foundation);
   }
 
+  /**
+   * Returns the state of the native control element, or undefined if the native control element is not present.
+   * @return {SelectionControlState|undefined}
```
Comment (acdvorak) [inline packages/mdc-radio/index.js]: `{SelectionControlState|undefined}` -> `{?SelectionControlState}`

`querySelector()` returns `null`, not `undefined`.

File: packages/mdc-radio/index.js [hunk 9]
```diff
@@ -84,15 +102,24 @@ export class MDCRadio extends MDCComponent {
...
+   * @private
+   */
   get nativeControl_() {
-    return this.root_.querySelector(MDCRadioFoundation.strings.NATIVE_CONTROL_SELECTOR);
+    const {NATIVE_CONTROL_SELECTOR} = MDCRadioFoundation.strings;
+    const el = /** @type {SelectionControlState|undefined} */ (
```
Comment (acdvorak) [inline packages/mdc-radio/index.js]: `{SelectionControlState|undefined}` -> `{?SelectionControlState}`

`querySelector()` returns `null`, not `undefined`.

**Comment by lynnjepsen** [comment]
Thanks for the review! Addressed comments. PTAL.

**Pull request merged**